src/BuildScriptGenerator/SdkStorageVersionProviderBase.cs

```diff
@@ -50,19 +50,26 @@ protected PlatformVersionInfo GetAvailableVersionsFromStorage(string platformNam
         {
             this.logger.LogDebug("Getting list of available versions for platform {platformName}.", platformName);
             var httpClient = this.HttpClientFactory.CreateClient("general");
-
             var sdkStorageBaseUrl = this.GetPlatformBinariesStorageBaseUrl();
-            var url = string.Format(SdkStorageConstants.ContainerMetadataUrlFormat, sdkStorageBaseUrl, platformName);
+            var url = string.Format(SdkStorageConstants.ContainerMetadataUrlFormat, sdkStorageBaseUrl, platformName, string.Empty);
             var blobList = httpClient.GetStringAsync(url).Result;
             var xdoc = XDocument.Parse(blobList);
+            var marker = xdoc.Root.Element("NextMarker").Value;
             var supportedVersions = new List<string>();
 
             var isStretch = string.Equals(this.commonOptions.DebianFlavor, OsTypes.DebianStretch, StringComparison.OrdinalIgnoreCase);
 
             var sdkVersionMetadataName = isStretch
                 ? SdkStorageConstants.LegacySdkVersionMetadataName
                 : SdkStorageConstants.SdkVersionMetadataName;
-
+            do
+            {
+                url = string.Format(SdkStorageConstants.ContainerMetadataUrlFormat, sdkStorageBaseUrl, platformName, marker);
+                var blobListFromNextMarker = httpClient.GetStringAsync(url).Result;
+                marker = XDocument.Parse(blobListFromNextMarker).Root.Element("NextMarker").Value;
+                xdoc.Descendants("Blobs").LastOrDefault().AddAfterSelf(XDocument.Parse(blobListFromNextMarker).Descendants("Blobs"));
```

Comment:
nit: would you mind adding a comment above this line describing what's occurring? It may not be obvious that we're adding the collection of blobs from the next page to the `xdoc` object that we iterate over later.